### PR TITLE
feat: `psycopg` asyncio support, `BigInteger` PK option, and binary GUIDs

### PR DESCRIPTION
**Pull request by JacobCoffee**
### Pull Request Checklist

[//]: # "Please review the [Litestar contribution guidelines](https://github.com/litestar-org/litestar/blob/main/CONTRIBUTING.rst) for this repository."

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

[//]: # "By submitting this issue, you agree to:"

[//]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

### Description

[//]: # "Please describe your pull request for new release changelog purposes"
Introduces the `BigIntPK` mixin for SQLAlchemy 2 contrib.
Adds `psycopg` tests for the sqlalchemy repository
Store GUIDs as binary on databases that don't have a native GUID type.

### Close Issue(s)

[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

### CODE REVIEW

File: docs/usage/security/abstract-authentication-middleware.rst [hunk 1]
```diff
@@ -55,7 +55,7 @@ example here lets say it's a pydantic model:
     Base = declarative_base()
 
 
-    class User(Base):
+    class User(UUIDBase):
```
Comment (cofin) [inline docs/usage/security/abstract-authentication-middleware.rst:58]: I don't think it gets changed here.

File: tests/contrib/sqlalchemy/test_dto.py [hunk 2]
```diff
@@ -321,10 +321,10 @@ async def test_dto_mapped_union(
 from litestar.contrib.sqlalchemy.dto import SQLAlchemyDTO
 from litestar.dto.factory import DTOConfig
 
-class Base(DeclarativeBase):
+class UUIDBase(DeclarativeBase):
```
Comment (cofin) [inline tests/contrib/sqlalchemy/test_dto.py:324]: And here.

**Comment by cofin** [comment]
Hey, overall, I think this is the way to go.  Based on some conversation today, I think we need to provide a non-uuid based PK as well.  

There are a few places in the documentations and examples where we are building a new Base from DeclarativeBase.  We don't want to update these examples.

**Comment by cofin** [comment]
I've broken the docs, somehow.

But this now adds in a custom `BigIntIdentity` SQL Alchemy data type.  `sqlite` only support auto-incrementing primary keys with an `Integer` type.  This type will use `BigInteger` where possible and fail back to `Integer` for `sqlite`.

**Comment by provinzkraut** [comment]
@cofin 

> I've broken the docs, somehow.

Fixed it! The classes were missing from `__all__` and there were some indentation issues with a list

**Comment by provinzkraut** [comment]
@cofin I think you broke it again with your force push :grimacing:

**Comment by cofin** [comment]
You weren't supposed to see that!

**Comment by cofin** [comment]
Closing this and re-opening it.

**Pull request closed without merging**